src/runtime/pydict.cs

```diff
@@ -22,6 +22,7 @@ public PyDict(IntPtr ptr) : base(ptr)
         {
         }
 
+        internal PyDict(BorrowedReference reference) : base(reference) { }
```

Comment:
This overload actually does incref internally (which makes sense, since to create a potentially long-lived `PyDict` object pointing to something we borrowed, we need to `IncRef`).